src/gui/editors/TimeLineWidget.cpp

```diff
@@ -271,12 +276,14 @@ void TimeLineWidget::paintEvent( QPaintEvent * )
 	int const x = m_xOffset + s_posMarkerPixmap->width() / 2 -
 			( ( static_cast<int>( m_begin * m_ppb ) / TimePos::ticksPerBar() ) % static_cast<int>( m_ppb ) );
 
+	// Double the interval between bar numbers until they are far enough appart
+	int barLabelInterval = 1;
+	while (barLabelInterval * m_ppb < MIN_BAR_LABEL_DISTANCE) { barLabelInterval *= 2; }
+
 	for( int i = 0; x + i * m_ppb < width(); ++i )
 	{
 		++barNumber;
-		// round m_ppb to the nearest power of two
-		const float ppbP2 = pow(2, round(log2(m_ppb)));
```

Comment:
I must say your calculation here is really clever, but I tried to make it a bit more understandable for a math noob like me.